### PR TITLE
TechDocs: Add docs by default to react ssr template

### LINKED ISSUE #2262: Extend the already existing template(s) (start with react ssr template) to include Documentation.

### PR DESCRIPTION
**Pull request by emmaindal**
## Hey, I just made a Pull Request!

This PR adds docs to the React SSR template so that you get docs by default when creating a new component. Later on it will be possible to select whether you want documentation or not in the create component flow.

Screen Shot 2020-09-03 at 10 49 09 AM
Screen Shot 2020-09-03 at 10 46 21 AM

closes: #2262

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+## {{ cookiecutter.component_id }}
```
Comment (stefanalund) [inline plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md:1]: Should this be only one `#`?

**Review by stefanalund — APPROVED** [review]

File: plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+{{ cookiecutter.description }}
+
+## Getting started
+
+Start write your documentation by adding more markdown (.md) files to this folder (/docs) or replace the content in this file.
```
Comment (OrkoHunter) [inline plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md:7]:
```suggestion
Start writing your documentation by adding more markdown (.md) files to this folder (/docs) or replace the content in this file.
```

File: plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md (hunk 1 above)
Comment (emmaindal) [reply plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md:1]:
```suggestion
# {{ cookiecutter.component_id }}
```

File: plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+For new pages to appear in the left hand navigation you need edit the `mkdocs.yml`
+file in root of your repo. The navigation can also link out to other sites.
+
+Alternatively, if there is no `nav` section in `mkdocs.yml`, a navigation section
+will be created for you. However, you will not be able to use alternate titles for
+pages, or include links to other sites.
+
+Note that MkDocs uses `mkdocs.yml`, not `mkdocs.yaml`, although both appear to work.
+See also <https://www.mkdocs.org/user-guide/configuration/>.
```
Comment (OrkoHunter) [inline plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md:24]: I struggled a bit when I wanted to create a "folding" navigation bar. I think we should link a very good example of `mkdocs.yml` here (if or when we have it).

**Review by OrkoHunter — APPROVED** [review]
Looks great!

File: plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md (hunk 3 above)
Comment (emmaindal) [reply plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md:24]: Good point! I think we can add more examples (on using our extensions etc as well) to this default docs content when it gets added to more templates. This is just a first step.

Comment (emmaindal) [reply plugins/scaffolder-backend/sample-templates/react-ssr-template/{{cookiecutter.component_id}}/docs/index.md:24]: Very soon we want this to be a optional step (but pre-selected) so you can "opt out" docs if you don't want it and at that point we can start adding more things.

**Pull request merged**